pyquil/tests/test_reference_density_simulator.py

```diff
@@ -309,3 +310,29 @@ def native_quil_to_executable(self, nq_program: Program):
         list(measure_observables(qc=qc_density, tomo_experiment=experiment_1q, n_shots=3000))
     except ValueError as e:
         assert str(e) != 'probabilities are not non-negative'
+
+
+def test_set_density():
+    # That is test the assigned state matrix in ReferenceDensitySimulator is persistent between
+    # rounds of run.
+    rho1 = np.array([[0.0, 0.0], [0.0, 1.0]])
+
+    # run prog
+    prog = Program(I(0))
+    ro = prog.declare('ro', 'BIT', 1)
+    prog += MEASURE(0, ro[0])
+
+    qc_density.qam.wf_simulator.set_density(rho1)
+
+    out = []
```

Comment:
`out = [qc_density.run(prog) for _ in range(0, 4)]`

Reply:
- [x] todo